File: demo/src/main/java/com/utcn/demo/entity/Content.java
package com.utcn.demo.entity;

import jakarta.persistence.*;

import java.time.LocalDateTime;

@Entity
@Table(name="content")
public class Content {
    @Id
    @Column(name="id")
    private Long contentId;

    @OneToOne
    @JoinColumn(name= "u_id",nullable = false)
    private User contentUser;

    @Column(name="c_text")
    private String contentText;

    @Column(name="c_date_time")
    private LocalDateTime contentDateTime;

    @Column(name="picture")
    private String contentPicture;

    public Content() {
    }

    public Content(Long contentId, User contentUser, String contentText, LocalDateTime contentDateTime, String contentPicture) {
        this.contentId = contentId;
        this.contentUser = contentUser;
        this.contentText = contentText;
        this.contentDateTime = contentDateTime;
        this.contentPicture = contentPicture;
    }

    public Long getContentId() {
        return contentId;
    }

    public void setContentId(Long contentId) {
        this.contentId = contentId;
    }

    public User getContentUser() {
        return contentUser;
    }

    public void setContentUser(User contentUser) {
        this.contentUser = contentUser;
    }

    public String getContentText() {
        return contentText;
    }

    public void setContentText(String contentText) {
        this.contentText = contentText;
    }

    public LocalDateTime getContentDateTime() {
        return contentDateTime;
    }

    public void setContentDateTime(LocalDateTime contentDateTime) {
        this.contentDateTime = contentDateTime;
    }

    public String getContentPicture() {
        return contentPicture;
    }

    public void setContentPicture(String contentPicture) {
        this.contentPicture = contentPicture;
    }
}

File: app/pages/content.ts
import {User} from "./user/user";

export class Content{


  private _contentId:number;

  private _contentUser:User;


  private _contentText:string;


  private _contentDateTime:string;


  private _contentPicture:string;


  constructor(contentId: number, contentUser: User, contentText: string, contentDateTime: string, contentPicture: string) {
    this._contentId = contentId;
    this._contentUser = contentUser;
    this._contentText = contentText;
    this._contentDateTime = contentDateTime;
    this._contentPicture = contentPicture;
  }

  get contentId(): number {
    return this._contentId;
  }

  set contentId(value: number) {
    this._contentId = value;
  }

  get contentUser(): User {
    return this._contentUser;
  }

  set contentUser(value: User) {
    this._contentUser = value;
  }

  get contentText(): string {
    return this._contentText;
  }

  set contentText(value: string) {
    this._contentText = value;
  }

  get contentDateTime(): string {
    return this._contentDateTime;
  }

  set contentDateTime(value: string) {
    this._contentDateTime = value;
  }

  get contentPicture(): string {
    return this._contentPicture;
  }

  set contentPicture(value: string) {
    this._contentPicture = value;
  }
}
